sub-packages/bionemo-scdl/tests/bionemo/scdl/util/test_torch_dataloader_utils.py

```diff
@@ -93,3 +93,306 @@ def test_dataloading_batch_size_three_works_with_collate(tmp_path, test_director
+    assert result['batch_size'] == 2
+    
+    # Check current cells sparse tensor
+    current_dense = result['current_cells'].to_dense()
+    expected_current = torch.tensor([[1.0, 2.0, 0.0, 0.0], [0.0, 4.0, 5.0, 6.0]])
+    assert torch.equal(current_dense, expected_current)
+    
+    # Check next cells sparse tensor
+    next_dense = result['next_cells'].to_dense()
+    expected_next = torch.tensor([[0.0, 0.0, 3.0], [7.0, 8.0, 0.0]])
+    assert torch.equal(next_dense, expected_next)
+
+
+def test_neighbor_collate_function_with_empty_cells():
+    """Test collate_neighbor_sparse_matrix_batch handles empty cells correctly."""
+    batch = [
+        {
+            'current_cell': ([1.0], [0]),
+            'next_cell': ([], []),  # Empty cell
+            'current_cell_index': 0,
+            'next_cell_index': 1
+        },
+        {
+            'current_cell': ([], []),  # Empty cell
+            'next_cell': ([2.0], [1]),
+            'current_cell_index': 2,
+            'next_cell_index': 3
+        }
+    ]
+    
+    result = collate_neighbor_sparse_matrix_batch(batch)
+    
+    # Check structure
+    assert result['batch_size'] == 2
+    assert result['current_cell_indices'] == [0, 2]
+    assert result['next_cell_indices'] == [1, 3]
+    
+    # Check current cells - first has data, second is empty
+    current_dense = result['current_cells'].to_dense()
+    expected_current = torch.tensor([[1.0], [0.0]])
+    assert torch.equal(current_dense, expected_current)
+    
+    # Check next cells - first is empty, second has data
+    next_dense = result['next_cells'].to_dense()
+    expected_next = torch.tensor([[0.0, 0.0], [0.0, 2.0]])
+    assert torch.equal(next_dense, expected_next)
+
+
+def test_neighbor_collate_function_all_empty():
+    """Test collate_neighbor_sparse_matrix_batch handles all empty cells."""
+    batch = [
+        {
+            'current_cell': ([], []),
+            'next_cell': ([], []),
+            'current_cell_index': 0,
+            'next_cell_index': 1
+        },
+        {
+            'current_cell': ([], []),
+            'next_cell': ([], []),
+            'current_cell_index': 2,
+            'next_cell_index': 3
+        }
+    ]
+    
+    result = collate_neighbor_sparse_matrix_batch(batch)
+    
+    assert result['batch_size'] == 2
+    assert result['current_cell_indices'] == [0, 2]
+    assert result['next_cell_indices'] == [1, 3]
+    
+    # Both tensors should have shape [2, 0] for all empty data
+    assert result['current_cells'].to_dense().shape == torch.Size([2, 0])
+    assert result['next_cells'].to_dense().shape == torch.Size([2, 0])
+
+
+def test_dataloading_neighbor_batch_with_real_data(tmp_path):
+    """Test neighbor collate function with real sample data."""
+    import os
+    from pathlib import Path
+    
+    # Check if sample neighbor data exists
+    sample_path = Path("/workspaces/bionemo-framework/scdl_data_local/adata_sample0_neighbors.h5ad")
```

Comment:
you can
1. Upload the data under "scdl/single-cell-test_data-neighbors-....."
2. load(scdl/single-cell-test_data-neighbors-....., source="pbss")

https://github.com/NVIDIA/bionemo-framework/blob/8d3d043cc7f4f61c309a1680a0c4e9b2e2e52f74/sub-packages/bionemo-geneformer/tests/bionemo/geneformer/scripts/test_pydantic_train.py#L92